src/include/sof/lib/notifier.h

```diff
@@ -29,6 +29,7 @@ enum notify_sync_id {
 	NOTIFIER_ID_SYNC_BUFFER_PRODUCE = 0,	/* struct buffer_cb_transact* */
 	NOTIFIER_ID_SYNC_BUFFER_CONSUME,	/* struct buffer_cb_transact* */
 	NOTIFIER_ID_SYNC_BUFFER_FREE,		/* struct buffer_cb_free* */
+	NOTIFIER_ID_SYNC_DMA_COPY,		/* struct dma_cb_data* */
```

Comment:
Could you also refactor irq_callback added in https://github.com/thesofproject/sof/commit/4839742c35f37f54af4ce6169cf3e591db49b2ad to use new notifier API?